src/moj/components/date-picker/date-picker.js

```diff
@@ -0,0 +1,948 @@
+  event.preventDefault();
+  const date = new Date(this.currentDate);
+  date.setFullYear(date.getFullYear() - 1, date.getMonth(), 1);
+  this.goToDate(date, focus);
+};
+
+/**
+ * Parse dataset
+ *
+ * Loop over an object and normalise each value using {@link normaliseString},
+ * optionally expanding nested `i18n.field`
+ *
+ * @param {{ schema: Schema }} Component - Component class
+ * @param {DOMStringMap} dataset - HTML element dataset
+ * @returns {Object} Normalised dataset
+ */
+Datepicker.prototype.parseDataset = function (schema, dataset) {
+  const parsed = {};
+
+  for (const [field, attributes] of Object.entries(schema.properties)) {
+    if (field in dataset) {
+      parsed[field] = dataset[field];
+    }
+  }
+
+  return parsed;
+};
+
+/**
+ * Config merging function
+ *
+ * Takes any number of objects and combines them together, with
+ * greatest priority on the LAST item passed in.
+ *
+ * @param {...{ [key: string]: unknown }} configObjects - Config objects to merge
+ * @returns {{ [key: string]: unknown }} A merged config object
+ */
+Datepicker.prototype.mergeConfigs = function (...configObjects) {
+  const formattedConfigObject = {};
+
+  // Loop through each of the passed objects
+  for (const configObject of configObjects) {
+    for (const key of Object.keys(configObject)) {
+      const option = formattedConfigObject[key];
+      const override = configObject[key];
+
+      // Push their keys one-by-one into formattedConfigObject. Any duplicate
+      // keys with object values will be merged, otherwise the new value will
+      // override the existing value.
+      if (typeof option === "object" && typeof override === "object") {
+        // @ts-expect-error Index signature for type 'string' is missing
+        formattedConfigObject[key] = this.mergeConfigs(option, override);
+      } else {
+        formattedConfigObject[key] = override;
+      }
+    }
+  }
+
+  return formattedConfigObject;
+};
+
+
+/**
+ *
+ * @param {HTMLElement} button
+ * @param {number} index
+ * @param {number} row
+ * @param {number} column
+ * @param {Datepicker} picker
+ * @constructor
+ */
+function DSCalendarDay(button, index, row, column, picker) {
+  this.index = index;
+  this.row = row;
+  this.column = column;
+  this.button = button;
+  this.picker = picker;
+
+  this.date = new Date();
+}
+
+DSCalendarDay.prototype.init = function () {
+  this.button.addEventListener("keydown", this.keyPress.bind(this));
+  this.button.addEventListener("click", this.click.bind(this));
+};
+
+/**
+ * @param {Date} day - the Date for the calendar day
+ * @param {boolean} hidden - visibility of the day
+ * @param {boolean} disabled - is the day selectable or excluded
+ */
+DSCalendarDay.prototype.update = function (day, hidden, disabled) {
+  let label = day.getDate();
+  let accessibleLabel = day.toLocaleDateString("en-GB", {
+    weekday: "long",
+    year: "numeric",
+    month: "long",
+    day: "numeric",
+  });
```

Comment:
It feels a bit weird that this is using the DateTimeFormat API, but elsewhere we're manually deriving it manually (`formattedDateHuman`).

A benefit of using the API is free localisation, but the labels etc. wouldn't match and it's hard-coded as en-GB anyway.